libs/elasticsearch-core/src/main/java/org/elasticsearch/bootstrap/JarHell.java

```diff
@@ -68,25 +67,22 @@ private JarHell() {}
     @SuppressForbidden(reason = "command line tool")
     public static void main(String args[]) throws Exception {
         System.out.println("checking for jar hell...");
-        checkJarHell();
+        checkJarHell(System.out::println);
         System.out.println("no jar hell found");
     }
 
     /**
      * Checks the current classpath for duplicate classes
      * @throws IllegalStateException if jar hell was found
      */
-    public static void checkJarHell() throws IOException, URISyntaxException {
+    public static void checkJarHell(Consumer<String> output) throws IOException, URISyntaxException {
```

Comment:
Can you update the Javadoc to explain this parameter?

Reply:
Certainly, I've pushed a commit that does this